triton.go

```diff
@@ -0,0 +1,18 @@
+package triton
+
+import (
+	"github.com/joyent/triton-go/authentication"
+)
+
+// Universal package used for defining configuration used across all client
+// constructors.
+
+// ClientConfig is a placeholder/input struct around the behavior of configuring
+// a client constructor through the implementation's runtime environment
+// (SDC/MANTA env vars).
+type ClientConfig struct {
+	Endpoint    string
```

Comment:
This. This caused a bug in signedURLs that I fixed here: https://github.com/gwydirsam/triton-go/commit/1740705ff9a9fe9912cb33fc84acb07595d49153